File: src/types/setor.ts
/**
 * Tipos relacionados a Setores
 */

import type { ID } from './common'

/**
 * Setor da empresa
 */
export interface Setor {
  id: ID
  codigo: string // "1", "2", "X", etc
  nome: string // Nome curto
  nomeCompleto?: string // Nome completo (opcional, API pode retornar null)
  cor: string // Cor de fundo (hex)
  corTexto: string // Cor do texto (hex)
  icone?: string // Ícone do setor (opcional)
  diretorId?: ID // FK para Colaborador responsável
  ordem: number // Ordem de exibição
}

/**
 * Subsetor (ex: Engenharia Civil, Engenharia Mecânica)
 */
export interface Subsetor {
  id: ID
  setorId: ID // FK para Setor pai
  nome: string
  cor?: string // Cor de fundo (opcional, API pode retornar null)
  corTexto?: string // Cor do texto (opcional, API pode retornar null)
}

/**
 * Dados para criar um novo setor
 */
export type SetorCreate = Omit<Setor, 'id'>

/**
 * Dados para atualizar um setor
 */
export type SetorUpdate = Partial<Omit<Setor, 'id'>>

/**
 * Dados para criar um novo subsetor
 */
export type SubsetorCreate = Omit<Subsetor, 'id'>

/**
 * Setor com contagem de colaboradores
 */
export interface SetorComContagem extends Setor {
  totalColaboradores: number
  colaboradoresPorNivel: Record<number, number>
}

File: backend/app/schemas/setor.py
"""
Schemas de Setor e Subsetor
"""
from pydantic import BaseModel


class SubsetorBase(BaseModel):
    nome: str
    cor: str | None = None
    cor_texto: str | None = "#1e293b"


class SubsetorCreate(SubsetorBase):
    setor_id: int


class SubsetorResponse(SubsetorBase):
    id: int
    setor_id: int

    class Config:
        from_attributes = True


class SetorBase(BaseModel):
    codigo: str = ""
    nome: str
    nome_completo: str | None = None
    cor: str = "#D3D3D3"
    cor_texto: str = "#1e293b"
    icone: str | None = None
    diretor_id: int | None = None
    ordem: int = 0


class SetorCreate(SetorBase):
    pass


class SetorUpdate(BaseModel):
    codigo: str | None = None
    nome: str | None = None
    nome_completo: str | None = None
    cor: str | None = None
    cor_texto: str | None = None
    icone: str | None = None
    diretor_id: int | None = None
    ordem: int | None = None


class SetorResponse(SetorBase):
    id: int
    subsetores: list[SubsetorResponse] = []

    class Config:
        from_attributes = True
